pkg/metaclient/kv.go

```diff
@@ -0,0 +1,47 @@
+package metaclient
+
+import "context"
+
+type Txn interface {
+	// Cache Ops in the Txn
+	Do(ops ...Op) Txn
+
+	// Commit tries to commit the transaction.
+	// Any Op error will cause entire txn rollback and return error
+	Commit(ctx context.Context) (*TxnResponse, error)
+}
+
+type KV interface {
+	// Put puts a key-value pair into metastore.
+	// Note that key,value can be plain bytes array and string is
+	// an immutable representation of that bytes array.
+	// To get a string of bytes, do string([]byte{0x10, 0x20}).
+	// When passed WithTTL(ttl), Put will insert a key with ttl duration.
+	// When passed WithRevision(revision), Put will be executed with corresponding revision, or do nothing on vice verse.
+	Put(ctx context.Context, key, val string, opts ...OpOption) (*PutResponse, error)
```

Comment:
why the opts is needed?

Reply:
Options carry some attributes in interface. In this case, options can carry the ttl and revision attributes for the key, to achieve key lifetime and idemptent.

Reply:
if using tidb instead of etcd,how to implement this interface

Reply:
IMHO, we can design a table with implicit column like revision(bigint unsigned) , startTs(timestamp), ttl(int) to support these attributes, so that `where revision=xxx` can help us achieve idempotent easily.

Reply:
Another way is to separate the `put` and `txn`, `txn` does the complex operation, such as setting a key with specific revision, for several reasons
- In many cases idempotent is not necessary for `put` operation. `put` with options makes the `put` operation has too complex semantics
- In DM and TiCDC, they use `etcd.Txn` for correctness serious scenario.